src/gui/codeeditors/qgscodeeditorhtml.cpp

```diff
@@ -76,3 +91,46 @@ void QgsCodeEditorHTML::initializeLexer()
   setLexer( lexer );
   runPostLexerConfigurationTasks();
 }
+
+QString QgsCodeEditorHTML::reformatCodeString( const QString &string )
+{
+  if ( !QgsPythonRunner::isValid() )
+  {
+    return string;
+  }
+  QString newText = string;
+  const QString definePrettify = QStringLiteral(
+                                   "def __qgis_prettify(html):\n"
+                                   "  try:\n"
+                                   "    from bs4 import BeautifulSoup\n"
+                                   "  except ImportError:\n"
+                                   "    return '_ImportError'\n"
```

Comment:
What about trying lxml if BeautifulSoup isn't available?

```
def __qgis_prettify(html):
    try:
        from bs4 import BeautifulSoup
        return BeautifulSoup(html, 'html.parser').prettify()
    except ImportError:
        try:
            from lxml import etree as LET
            from io import BytesIO
            parser = LET.XMLParser(remove_blank_text=True)
            file_obj = BytesIO(html.encode('UTF-8'))
            try:
                root = LET.parse(file_obj, parser)
            except LET.XMLSyntaxError:
                return '_SyntaxError'

            return LET.tostring(root, pretty_print=True).decode('UTF-8')
        except ImportError:
            return '_ImportError'
```

Reply:
(this would also require checking if the result is "_SyntaxError" and leaving the text unchanged if so, or showing some descriptive warning)